### PR TITLE
Upgrade to Python 2.7.10

### PR DESCRIPTION
**Pull request by msmolens**
This change upgrades Slicer's Python from version 2.7.3 to version 2.7.10. The newer Python version fixes a number of build warnings seen when building Slicer with Python 2.7.3 in Visual Studio 2013, like:

    C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xtgmath.h(190): warning C4273: '_hypot' : inconsistent dll linkage [C:\dev\SN\VTKv6-build\Wrapping\PythonCore\vtkWrappingPythonCore.vcxproj]
        C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(538) : see previous definition of '_hypot'

and:

    C:\dev\SN\python-2.7.3\include\pymath.h(22): warning C4273: 'round' : inconsistent dll linkage [C:\dev\SN\python-build\CMakeBuild\libpython\libpython-shared.vcxproj]
        C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(516) : see previous definition of 'round'

This change also updates the following external packages:
- python-cmake-buildsystem: to support building Python 2.7.10 on all platforms
- CTK/PythonQt: to fix crashes on exit in Python/PythonQt with Python >2.7.3

### CODE REVIEW

File: SuperBuild/External_python.cmake [hunk 1]
```diff
@@ -113,6 +113,7 @@ if((NOT DEFINED PYTHON_INCLUDE_DIR
       -DBUILD_SHARED:BOOL=ON
       -DBUILD_STATIC:BOOL=OFF
       -DUSE_SYSTEM_LIBRARIES:BOOL=OFF
+      -DSRC_DIR:PATH=${python_SOURCE_DIR}
```
Comment (jcfr) [inline SuperBuild/External_python.cmake]: You could also set `-DDOWNLOAD_SOURCES:BOOL=OFF` and `-DINSTALL_WINDOWS_TRADITIONAL:BOOL=OFF`

That way, the configuration/build of python would be more robust to change in default option. See https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/commit/d40eb36424b264f5d97fe582479f181e00540dde

File: SuperBuild/python_patch.cmake [hunk 2]
```diff
@@ -4,7 +4,7 @@ set(out_msvc9compiler ${PYTHON_SRC_DIR}/Lib/distutils/msvc9compiler.py)
 
 #
 # This custom version of 'msvc9compiler.py' has been made by applying
-# the following patches to the version of 'msvc9compiler.py' shipped with Python-2.7.3.tgz:
+# the following patches to the version of 'msvc9compiler.py' shipped with Python-2.7.10.tgz:
```
Comment (jcfr) [inline SuperBuild/python_patch.cmake]: Does this mean that the file `python_patched_msvc9compiler.py` has not changed between `2.7.3` and `2.7.10` ?

**Comment by jcfr** [comment]
That is all very exciting ! Thanks for working on this. 

Before of the few remarks, it is getting close :) 

Could you also mention in the commit message on which plaform you manually tested and confirmed it works.

File: SuperBuild/python_patch.cmake (hunk 2 above)
Comment (msmolens) [inline SuperBuild/python_patch.cmake]: That's correct:

    $ md5sum ./Python-2.7.3/Lib/distutils/msvc9compiler.py ./Python-2.7.10/Lib/distutils/msvc9compiler.py
    908f2eefe08c08c33a9e97558c5c2d93 *./Python-2.7.3/Lib/distutils/msvc9compiler.py
    908f2eefe08c08c33a9e97558c5c2d93 *./Python-2.7.10/Lib/distutils/msvc9compiler.py

Comment (jcfr) [inline SuperBuild/python_patch.cmake]: Great. Include this in the commit message too :)

**Pull request closed without merging**